test/ttsafe_develop.c

```diff
@@ -99,26 +103,42 @@ tts_develop_api_thr_2(void *_udata)
 void
 tts_develop_api(const void H5_ATTR_UNUSED *params)
 {
+    hid_t                   def_fapl = H5I_INVALID_HID;
+    hid_t                   vol_id   = H5I_INVALID_HID;
     H5TS_thread_t           thread_1, thread_2;
     H5TS_barrier_t          barrier;
     unsigned                lock_count = UINT_MAX;
     bool                    acquired   = false;
     tts_develop_api_udata_t udata;
     unsigned                api_count_1 = 0, api_count_2 = 0;
+    int                     is_native;
     herr_t                  result;
 
-    /* Check that API count increases with each API call */
-    result = H5TSmutex_get_attempt_count(&api_count_1);
-    CHECK_I(result, "H5TSmutex_get_attempt_count");
+    def_fapl = H5Pcreate(H5P_FILE_ACCESS);
+    CHECK(def_fapl, H5I_INVALID_HID, "H5Pcreate");
+
+    result = H5Pget_vol_id(def_fapl, &vol_id);
+    CHECK(result, FAIL, "H5Pget_vol_id");
+
+    is_native = H5VL__is_native_connector_test(vol_id);
+    CHECK(is_native, FAIL, "H5VL__is_native_connector_test");
+
+    if (is_native) {
```

Comment:
Pass-through VOL connectors increment the API counter by various amounts, so just skip this portion of the test.